src/listeners/command-handler/CoreCommandAccepted.ts

```diff
@@ -12,6 +12,11 @@ export class CoreListener extends Listener<typeof Events.CommandAccepted> {
 		const args = await command.preParse(message, parameters, context);
 		try {
 			message.client.emit(Events.CommandRun, message, command, { ...payload, args });
+			if (command.typing || (command.typing === undefined && command.container.client.options.typing)) {
+				message.channel.sendTyping().catch((e) => {
+					throw e;
+				});
+			}
```

Comment:
This entire thing should be a Listener that disables itself when the option is set to false.

This way, you also skip a check that's done all the time.

Reply:
Do you mean a Listener in `src/lib/listeners/command-handler/`, e.g. `CoreCommandTyping.ts`? I'd like to ask if this is what you had in mind, because I am not entirely sure if I understood correctly. 😅 bitomic/framework@6bf4cdf1

Still, tested. It doesn't run when `SapphireClientOptions#typing` is `false` or `SapphireClientOptions#typing` is `true` but `CommandOptions#typing` is `false`.

Reply:
A piece has an `enabled` option that can be set to `false`

Reply:
@Favna like this? or is there something wrong with that way? 

https://github.com/bitomic/framework/blob/343008d4988b29983c65d3665fedd41fa094287f/src/listeners/command-handler/CoreCommandTyping.ts#L10